internal/git/url.go

```diff
@@ -60,7 +65,7 @@ func parseURL(rawURL string) (u *url.URL, err error) {
 }
 
 // Extract GitHub repository information from a git remote URL.
-func repoInfoFromURL(u *url.URL) (host string, owner string, repo string, err error) {
+func RepoInfoFromURL(u *url.URL) (host string, owner string, name string, err error) {
```

Comment:
This function has 4 return values, which makes it a poor public API in my view.

A more useful function than the current `ParseURL` + `RepoInfoFromURL` might be a single method that parses a string URL into `(Repository, error)` return values. What do you think?

Reply:
Agreed, this and `ParseURL` are the internal helper functions for the `Parse` function defined below in `pkg/repository/repository.go` and are not exposed to end users of the package.

Reply:
Ahh, completely missed the fact that this was in an internal package 🤦